atomic-design.md

```diff
@@ -0,0 +1,244 @@
+which will make even bigger pieces and so on until we arrive at our deliverable.
+
+Let's illustrate this with a real world example. If I were going to make a house I would need 
+several components. At a bare minimum would need a foundation, some walls, and a roof.  I would 
+also like to include items such as plumbing, electricity and gas.
+
+Each of these components are made of smaller components.  The walls are going to have studs, 
+drywall, paint, and windows.  Studs are going to need wood and nails.
+
+Some of these components by themselves are not very useful.  A nail by itself does nothing.  With 
+some wood, it can attach to a foundation and serve to complete my house.  I cannot have a house 
+without these smaller pieces.
+
+In web development we want to make a web page.  We can break a page down into components, and 
+those into smaller components to achieve our goals.  If done right, we have reusable components 
+that we can use in other places to give everything a cohesive look and feel, and predictable behavior.
+
+The ideas of Atomic Design help us do that.
+
+
+##Components
+
+Components in software development would be the nails and studs in the analogy above.  They can be 
+combined with other components to make bigger components.  With good design, they can be reused.  
+This allows us to write cleaner code, and less code.  Less code is a great way to reduce bugs and 
+problems in code.
+
+For example, a carpenter will often have more than one nail, in fact he will have an entire box of 
+nails.  If he needs a finishing nail, he will often reach into this container and grab one, 
+without looking.  Why? Because to him it doesn't matter what nail he grabs, as long as it is a finishing nail.
+
+We should write our components in the same way.   
+
+For example, let's start with a simple a component.
+
+     <input id="message-displayer">
+     <button onclick="document.getElementById('message-displayer').value = 'The button was clicked'">
+        Click Me
+      </button>
+
+This doesn't do much.  I click the button and the message "The button clicked" displays in the text box.  The potential for reuse is almost 0.
+
+However, with a few changes and some abstraction can do:
+
+    <input v-model="message" >
+    <button v-on:click="$emit('button-clicked')">
+        Click Me
+    </button>
+
+The above example uses Vue JS as an example.  But in simple terms it
+ 1. It will display whatever message is passed by the parent component
+ 2. The button will emit an event when clicked, and the parent can handle the event as it sees fit. 
+ It can change the message in the box, or something completely different.
+ 
+ The code is now much more reusable. Any where on the page that needs a button and a text field can 
+ use this.
+
+##Why Vue?
+
+First off, you do need Vue, or any framework to make reusable components, and use the principles of 
```

Comment:
I think you mean `you do not nee`

Reply:
Good catch!  I am fixing that as we speak.